app/lib/blazer_admin_constraint.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class BlazerAdminConstraint
+  def matches?(request)
+    signin_user = UserSession.load_from_session(request.session)
+    return false if signin_user.blank?
+
+    admin_user = User.admins.kept.find_by(email: signin_user.email)
+    admin_user.present? && authorised_for_blazer?(signin_user)
+  end
+
+  private
+
+  def authorised_for_blazer?(signin_user)
+    User.with_blazer_access.find_by(email: signin_user.email).present?
+  end
```

Comment:
```suggestion
    blazer_user = User.with_blazer_access.kept.find_by(email: signin_user.email)
    blazer_user.present?
  end
```

Reply:
Thanks! This is a lot neater, i've updated in this commit: https://github.com/DFE-Digital/publish-teacher-training/pull/4277/commits/c123a3516080f02deb075b3f3fbe690761662a60

Had to open a new PR.